test/framework/management_cluster.go

```diff
@@ -158,3 +166,87 @@ func WaitForDeploymentsAvailable(ctx context.Context, input WaitForDeploymentsAv
 
 	}, intervals...).Should(BeTrue(), "Deployment %s/%s failed to get status.Available = True condition", input.Deployment.GetNamespace(), input.Deployment.GetName())
 }
+
+// CreateNamespaceInput is the input type for CreateNamespace.
+type CreateNamespaceInput struct {
+	Creator Creator
+	Name    string
+}
+
+// CreateNamespace is used to create a namespace object.
+func CreateNamespace(ctx context.Context, input CreateNamespaceInput, intervals ...interface{}) *corev1.Namespace {
+	ns := &corev1.Namespace{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: input.Name,
+		},
+	}
+	By(fmt.Sprintf("Creating namespace %s", input.Name))
+	Eventually(func() error {
+		return input.Creator.Create(context.TODO(), ns)
+	}, intervals...).Should(Succeed())
+
+	return ns
+}
+
+// DeleteNamespaceInput is the input type for DeleteNamespace.
+type DeleteNamespaceInput struct {
+	Deleter Deleter
+	Name    string
+}
+
+// DeleteNamespace is used to delete namespace object.
+func DeleteNamespace(ctx context.Context, input DeleteNamespaceInput, intervals ...interface{}) {
+	ns := &corev1.Namespace{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: input.Name,
+		},
+	}
+	By(fmt.Sprintf("Deleting namespace %s", input.Name))
+	Eventually(func() error {
+		return input.Deleter.Delete(context.TODO(), ns)
+	}, intervals...).Should(Succeed())
+}
+
+// WatchNamespaceEventsInput is the input type for WatchNamespaceEvents.
+type WatchNamespaceEventsInput struct {
+	ClientSet *kubernetes.Clientset
+	Namespace string
+	LogPath   string
+}
+
+// WatchNamespaceEvents creates a watcher that streams namespace events into a file namespace
+func WatchNamespaceEvents(ctx context.Context, input WatchNamespaceEventsInput) {
+	logFile := path.Join(input.LogPath, "resources", input.Namespace, "events.log")
```

Comment:
To help troubleshoot issues if a consumer fails to pass in required values for input:
```suggestion
	Expect(input.ClientSet).NotTo(BeNil(), "input.ClientSet is required for WatchNamespaceEvents")
	Expect(input.Namespace).NotTo(BeEmpty(), "input.Namespace is required for WatchNamespaceEvents")
	Expect(input.LogPath).NotTo(BeEmpty(), "input.LogPath is required for WatchNamespaceEvents")
	logFile := path.Join(input.LogPath, "resources", input.Namespace, "events.log")
```

Reply:
done!